core/src/main/java/org/infinispan/notifications/cachelistener/annotation/TopologyChanged.java

```diff
@@ -13,8 +13,9 @@
 
 /**
  * This annotation should be used on methods that need to be notified when the {@link ConsistentHash} implementation
- * in use by the {@link DistributionManager} changes due to a change in cluster topology.  This is only fired
- * in a {@link org.infinispan.configuration.cache.CacheMode#DIST_SYNC} or {@link org.infinispan.configuration.cache.CacheMode#DIST_ASYNC} configured cache.
+ * in use by the {@link DistributionManager} changes due to a change in cluster topology.
+ * This is not fired for {@link org.infinispan.configuration.cache.CacheMode#LOCAL} nor {@link org.infinispan.configuration.cache.CacheMode#INVALIDATION_ASYNC}/
```

Comment:
```suggestion
 * This is not fired for {@link org.infinispan.configuration.cache.CacheMode#LOCAL} caches.
```

`TopologyChanged` is now fired for all clustered caches, only `DataRehashed` is not fired for invalidation caches.